spec/DefinedSchemas.spec.js

```diff
@@ -0,0 +1,641 @@
+        },
+        server.config
+      ).execute();
+
+      schema = await new Parse.Schema('Test').get();
+      expect(schema.fields.aField).toEqual({ type: 'String', required: true });
+
+      await object.fetch({ useMasterKey: true });
+      expect(object.get('aField')).toEqual('Hello');
+    });
+  });
+
+  describe('Indexes', () => {
+    it('should create new indexes', async () => {
+      const server = await reconfigureServer();
+
+      const indexes = { complex: { createdAt: 1, updatedAt: 1 } };
+
+      const schemas = {
+        schemas: [{ className: 'Test', fields: { aField: { type: 'String' } }, indexes }],
+      };
+      await new DefinedSchemas(schemas, server.config).execute();
+
+      let schema = await new Parse.Schema('Test').get();
+      cleanUpIndexes(schema);
+      expect(schema.indexes).toEqual(indexes);
+
+      indexes.complex2 = { createdAt: 1, aField: 1 };
+      await new DefinedSchemas(schemas, server.config).execute();
+      schema = await new Parse.Schema('Test').get();
+      cleanUpIndexes(schema);
+      expect(schema.indexes).toEqual(indexes);
+    });
+    it('should re create changed indexes', async () => {
+      const server = await reconfigureServer();
+
+      let indexes = { complex: { createdAt: 1, updatedAt: 1 } };
+
+      let schemas = { schemas: [{ className: 'Test', indexes }] };
+      await new DefinedSchemas(schemas, server.config).execute();
+
+      indexes = { complex: { createdAt: 1 } };
+      schemas = { schemas: [{ className: 'Test', indexes }] };
+
+      // Change indexes
+      await new DefinedSchemas(schemas, server.config).execute();
+      let schema = await new Parse.Schema('Test').get();
+      cleanUpIndexes(schema);
+      expect(schema.indexes).toEqual(indexes);
+
+      // Update
+      await new DefinedSchemas(schemas, server.config).execute();
+      schema = await new Parse.Schema('Test').get();
+      cleanUpIndexes(schema);
+      expect(schema.indexes).toEqual(indexes);
+    });
+
+    it('should delete removed indexes', async () => {
+      const server = await reconfigureServer();
+
+      let indexes = { complex: { createdAt: 1, updatedAt: 1 } };
+
+      let schemas = { schemas: [{ className: 'Test', indexes }] };
+      await new DefinedSchemas(schemas, server.config).execute();
+
+      indexes = {};
+      schemas = { schemas: [{ className: 'Test', indexes }] };
+      // Change indexes
+      await new DefinedSchemas(schemas, server.config).execute();
+      let schema = await new Parse.Schema('Test').get();
+      cleanUpIndexes(schema);
+      expect(schema.indexes).toBeUndefined();
+
+      // Update
+      await new DefinedSchemas(schemas, server.config).execute();
+      schema = await new Parse.Schema('Test').get();
+      cleanUpIndexes(schema);
+      expect(schema.indexes).toBeUndefined();
+    });
+    xit('should keep protected indexes', async () => {
```

Comment:
Here you can keep xit, this test is flaky, we need to investigate deeper later on schema system, or make the test flaky proof (with a while system). This test ensure that core default indexes on some classes are never removed even if the index is not mentioned into the schema indexes field.

Reply:
I think the preservation of required indexes is already done internally.

Ill comment out the test for now.